tensorflow/python/data/benchmarks/benchmark_base.py

```diff
@@ -31,12 +31,137 @@
 class DatasetBenchmarkBase(test.Benchmark):
   """Base class for dataset benchmarks."""
 
+  def _run_eager_benchmark(self, op_or_dataset, iters, warmup):
+    """Benchmark in eager mode.
+
+    Runs the op `iters` times. In each iteration, the benchmark measures
+    the time it takes to go execute the op.
+
+    Args:
+      op_or_dataset: The tf op or tf.data Dataset to benchmark.
+      iters: Number of times to repeat the timing.
+      warmup: If true, warms up the session caches by running an untimed run.
+
+    Returns:
+      A float, representing the median time (with respect to `iters`)
+      it takes for the op to be executed `iters` num of times.
+    """
+
+    deltas = []
+    if not context.executing_eagerly():
+      raise RuntimeError(
+          "Eager mode benchmarking is not supported in graph mode.")
+
+    for _ in range(iters):
+      if warmup:
+        iterator = iter(op_or_dataset)
+        next(iterator)
+
+      iterator = iter(op_or_dataset)
+      start = time.time()
+      next(iterator)
+      end = time.time()
+      deltas.append(end - start)
+    return np.median(deltas)
+
+  def _run_graph_benchmark(self,
+                           op_or_dataset,
+                           iters,
+                           warmup,
+                           session_config,
+                           is_dataset=True):
+    """Benchmarks the op in graph mode.
+
+    Runs the op `iters` times. In each iteration, the benchmark measures
+    the time it takes to go execute the op.
+
+    Args:
+      op_or_dataset: The tf op or tf.data Dataset to benchmark.
```

Comment:
ditto, replace `op_or_dataset` with `iterable`

Reply:
addressed in the commit